policy/test_overlayfs.te

```diff
@@ -0,0 +1,169 @@
+#################################
+#
+# Policy for testing overlayfs file system support
+#
+
+# Domain for mounter process.
+type test_overlay_mounter_t, testdomain;
+domain_type(test_overlay_mounter_t)
+
+# Domain for client process.
+type test_overlay_client_t, testdomain;
+domain_type(test_overlay_client_t)
+
+# Domain for server process.
+type test_overlay_server_t, testdomain;
+domain_type(test_overlay_server_t)
+
+type test_overlay_files_ro_t;
+files_type(test_overlay_files_ro_t)
+
+type test_overlay_files_rwx_t;
+files_type(test_overlay_files_rwx_t)
+
+type test_overlay_mounter_files_t;
+files_type(test_overlay_mounter_files_t)
+
+type test_overlay_whiteout_files_t;
+files_type(test_overlay_whiteout_files_t)
+
+type test_overlay_files_noaccess_t;
+files_type(test_overlay_files_noaccess_t)
+
+#
+# Mounter policy
+#
+allow test_overlay_mounter_t self:dir list_dir_perms;
+allow test_overlay_mounter_t self:file read_file_perms;
+allow test_overlay_mounter_t self:capability { sys_admin dac_override dac_read_search };
+
+unconfined_runs_test(test_overlay_mounter_t)
+
+kernel_read_proc_symlinks(test_overlay_mounter_t)
+kernel_request_load_module(test_overlay_mounter_t)
+kernel_search_proc(test_overlay_mounter_t)
+ 
+fs_relabelfrom_xattr_fs(test_overlay_mounter_t)
+fs_mount_xattr_fs(test_overlay_mounter_t)
+
+corecmd_shell_entry_type(test_overlay_mounter_t)
+corecmd_exec_bin(test_overlay_mounter_t)
+
+userdom_read_inherited_user_tmp_files(test_overlay_mounter_t)
+userdom_search_admin_dir(test_overlay_mounter_t)
+userdom_search_user_home_content(test_overlay_mounter_t)
+
+mount_exec(test_overlay_mounter_t)
+mount_rw_pid_files(test_overlay_mounter_t)
+
+fs_getattr_xattr_fs(test_overlay_mounter_t)
+
+selinux_getattr_fs(test_overlay_mounter_t)
+
+#
+# Mounter needs to be able to mount on r/o and r/w/x directories and files
+# relabelfrom/relableto can probably be removed if fscreate fixes work.
```

Comment:
Actually, just the comment is wrong - you don't allow mounter relabelfrom/relabelto the files, only to the filesystem (for context= mount).  So we can just drop the comment.